### ISSUE TITLE
downstream input and output

### ISSUE THREAD

**Issue by liu-x-p** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#49]
Hi, I studied this code for days, and have some questions for it.
When training the downstream model, when I set _phone_path_ to be _cpc_phone_, dataloaders will load features and labels. For example, the shape of labels is [12,1544], and for features, that is [12,1544,160].
12 is batch_size, maybe 1544 means the loaded feature is consist of 1544 phones(after zero padding?), and each 160-dim feature represents one phone. After the upstream model, it becomes a [12,1544,768] tensor which is the output of transformer.
Do I understand correctly?

**Comment by andi611** [comment]
Hi,

the shape of the label is `(batch_size, seq_len)`,
the shape of the feature is `(batch_size, seq_len, feature_dim)`.

> maybe 1544 means the loaded feature is consist of 1544 phones(after zero padding?)

The 1544 is the sequence length of that particular phone sequence, 
and also the sequence length of that particular feature sequence.

> and each 160-dim feature represents one phone.

Each 160-dim acoustic feature may **NOT** represent one phone, since each 160-dim is just a single acoustic frame, a phone usually spans over several frames.

> After the upstream model, it becomes a [12,1544,768] tensor which is the output of transformer.

Yes.

**Comment by liu-x-p** [comment]
Thank you for your reply. So does the seq_len mean the number of frames?

**Comment by andi611** [comment]
> Thank you for your reply. So does the seq_len mean the number of frames?

Yes!

**Comment by liu-x-p** [comment]
Thanks! Got it!
